src/validator.js

```diff
@@ -471,4 +474,34 @@ export default class Validator {
       return ARCH_DEFAULT;
     }
   }
+
+  _markEmptyFiles(addonMetadata) {
+    var emptyFiles = [];
+    var promises = [];
+
+    return this.io.getFiles()
+      .then((files) => {
+        for (let filename in files) {
+          promises.push(this._isFileEmpty(filename, emptyFiles));
+        }
+
+        return Promise.all(promises);
+      })
+      .then(() => {
+        addonMetadata.emptyFiles = emptyFiles;
+        return addonMetadata;
+      });
+  }
+
+  _isFileEmpty(filename, emptyFiles) {
+    return this.io.getFile(filename)
+      .then((contents) => {
+        // If this file is empty, we want to flag it as such.
+        if (contents.length === 0) {
```

Comment:
Can we determine from size metadata in both the XPI + file system cases? I also wonder if we also need a check to see if a file is just whitespace too?